diff --git a/ck.py b/ck.py
index 6b125a8..b4a3948 100755
--- a/ck.py
+++ b/ck.py
@@ -61,6 +61,7 @@ UI_INFO = """
 class CustomKey:
     def __init__(self):
         self.widgets = []
+        self.ids = {}
         self.editing = False
         ####Main Menu
         action_group = Gtk.ActionGroup("my_actions")
@@ -401,6 +402,8 @@ class CustomKey:
         return item_factory.get_widget("<main>")
 
     def move_key(self, widget, event, key):
+        #print event.get_click_count()[0]
+        #print dir(event)
         if not self.editing:
             self.evpos = self.save[key][0:2]
             self.evpos[0] += int(event.x - 25)
@@ -464,7 +467,9 @@ class CustomKey:
             b.set_border_width(1)
             b.set_size_request(self.save[self.cont][2], self.save[self.cont][3])
             b.connect('button-press-event', self.onButtonPress, self.cont, self.menu)
-            b.connect('motion-notify-event', self.move_key, self.cont)
+            b.connect("button-release-event", self.onButtonRelease, self.cont)
+            ####b.connect('motion-notify-event', self.move_key, self.cont)
+
             #b.connect_object("event", self.onButtonRightClick, self.menu)
             #self.menu_optionEditProperties.connect("button-press-event",
              #                                      self.editButton,
@@ -476,7 +481,11 @@ class CustomKey:
         self.cont += 1
 
     def onButtonPress(self, widget, event, key, menu):
+        print event.button
+        self.ids[key] = widget.connect('motion-notify-event', self.move_key, key)
+
         if event.type == Gdk.EventType._2BUTTON_PRESS and not self.editing:
+
             self.onButtonDoubleClick(widget, event, key)
 
         elif event.type == Gdk.EventType.BUTTON_PRESS and event.button == 3:
@@ -492,6 +501,8 @@ class CustomKey:
             self.evpos[1] += int(event.y - 25)
 
     def onButtonRelease(self, widget, event, key):
+        widget.disconnect(self.ids[key])
+        self.ids.pop(key)
         #print "br"
         self.pos = self.save[key][0:2]
         if (self.evpos[0] != self.pos[0] and
diff --git a/customWidgets.py b/customWidgets.py
index 9b60f82..6ccd885 100644
--- a/customWidgets.py
+++ b/customWidgets.py
@@ -27,7 +27,11 @@ class ResizableEventBox(Gtk.EventBox):
         self.modify_bg(Gtk.StateFlags.NORMAL, Gdk.Color(red=65535,
                                                         green=65535,
                                                         blue=65535))
-        self.set_events(Gdk.EventMask.POINTER_MOTION_MASK)
+        self.set_events(Gdk.EventMask.POINTER_MOTION_MASK |
+                        Gdk.EventMask.BUTTON_MOTION_MASK |
+                        Gdk.EventMask.BUTTON1_MOTION_MASK |
+                        Gdk.EventMask.BUTTON2_MOTION_MASK |
+                        Gdk.EventMask.BUTTON3_MOTION_MASK)
         self.connect("button-press-event", self.enableResize)
         self.connect("leave-notify-event", self.set_focus_out)
         self.connect("motion-notify-event", self.set_focus_in)
@@ -138,6 +142,7 @@ class ResizableEventBox(Gtk.EventBox):
 
     def enableResize(self, widget, event):
         print "this works"
+        print event.button
         self.typec = 1
         if self.showResize:
             return
